### PR TITLE
Fix more become plugins

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Similar to #704.

Also adds more tests.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
dzdo
pbrun

### CODE REVIEW

File: plugins/become/ksu.py [hunk 1]
```diff
@@ -13,7 +13,6 @@
     options:
         become_user:
             description: User you 'become' to execute the task
-            default: ''
```
Comment (felixfontein) [inline plugins/become/ksu.py:16]: Required values need no default.

File: plugins/become/dzdo.py [hunk 2]
```diff
@@ -90,7 +90,6 @@ def build_become_command(self, cmd, shell):
             flags = '%s -p "%s"' % (flags.replace('-n', ''), self.prompt)
 
         user = self.get_option('become_user')
```
Comment (goneri) [inline plugins/become/dzdo.py:92]: More a matter of style, but I would prefer to use a variable called `become_user` and another `user`, like in doas.py. This way we don't have to change the purpose and type of the `user` in the middle of the code.

Comment (felixfontein) [reply plugins/become/dzdo.py]: I've adjusted the code accordingly.

**Comment by felixfontein** [comment]
The remaining tests were hanging and aren't relevant.

**Comment by felixfontein** [comment]
@goneri thanks a lot for reviewing this!

**Pull request merged**